test/controllers/api/v2/organizations_controller_test.rb

```diff
@@ -89,6 +89,95 @@ def test_create_duplicate_name
       assert_response :unprocessable_entity
     end
 
+    def test_create_with_auto_label
+      Organization.any_instance.stubs(:redhat_repository_url)
+      Organization.any_instance.stubs(:default_content_view).returns(OpenStruct.new(id: 1))
+      Organization.any_instance.stubs(:library).returns(OpenStruct.new(id: 10))
+      name = "Organization With Label"
+      assert_sync_task ::Actions::Katello::Organization::Create do |org|
+        org.name.must_equal name
+        assert org.valid?
+        org.label.wont_be_empty
+        org.stubs(:reload)
+      end
+      post :create, params: { :organization => {:name => name} }
+      assert_response :success
+    end
+
+    def test_create_identical_name_and_label
+      Organization.any_instance.stubs(:redhat_repository_url)
+      Organization.any_instance.stubs(:default_content_view).returns(OpenStruct.new(id: 1))
+      Organization.any_instance.stubs(:library).returns(OpenStruct.new(id: 10))
+      name = "organization_with_name_and_label_identical"
+      assert_sync_task ::Actions::Katello::Organization::Create do |org|
+        org.name.must_equal name
+        org.label.must_equal name
+        assert org.valid?
+        org.stubs(:reload)
+      end
+      post :create, params: { :organization => {:name => name, :label => name} }
+      assert_response :success
+    end
+
+    def test_create_with_name_and_label
+      Organization.any_instance.stubs(:redhat_repository_url)
+      Organization.any_instance.stubs(:default_content_view).returns(OpenStruct.new(id: 1))
+      Organization.any_instance.stubs(:library).returns(OpenStruct.new(id: 10))
+
+      name = "Organization With Label"
+      label = "org_with_label"
+      assert_sync_task ::Actions::Katello::Organization::Create do |org|
+        org.name.must_equal name
+        org.label.must_equal label
+        assert org.valid?
+        org.stubs(:reload)
+      end
+      post :create, params: { :organization => {:name => name, :label => label} }
+      assert_response :success
+    end
+
+    def test_create_with_name_label_description
+      Organization.any_instance.stubs(:redhat_repository_url)
+      Organization.any_instance.stubs(:default_content_view).returns(OpenStruct.new(id: 1))
+      Organization.any_instance.stubs(:library).returns(OpenStruct.new(id: 10))
+
+      name = "Organization With Name Label Description"
+      label = "organization_with_name_label_description"
```

Comment:
what is the point of specifying a label in this test, since it is covered in the previous test?

Reply:
This test create an organization using the combination of fields name, label and description the previous test only label and name.

Notes: 
- This is the same test as in robottelo https://github.com/SatelliteQE/robottelo/blob/master/tests/foreman/api/test_organization.py#L155 
- We are not trying to make a triage
- We already made many simplification when splitting robottelo  test between the two project eg: foreman/katello and left here only those related to katello.